app/main.py

```diff
@@ -1 +1,17 @@
-# write your code here
+def create_report(data_file_name: str, report_file_name: str):
+
+    supply = 0
+    buy = 0
+
+    with open(data_file_name, "r") as data_file:
+        with open(report_file_name, "w") as report_file:
```

Comment:
It's better not to use nested context managers here. You can first read data from `data_file` and only then open and write to report